client/views/7-ShowContactProfileView.js

```diff
@@ -0,0 +1,299 @@
+
+import HeadersActions from './../components/HeadersActions';
+
+const canOpenURL = (url) => {
+  Linking.canOpenURL(url).then(supported => {
+    if (!supported) console.log('Can\'t handle url: ' + url);
+    else return Linking.openURL(url);
+  }).catch(err => console.error('An error occurred', err));
+};
+
+class ContactProfileView extends Component {
+
+  static navigationOptions = {
+    headerBackgroundTransitionPreset: 'toggle',
+    title: 'View Profile',
+    headerRight: (
+      <HeadersActions></HeadersActions>),
+    headerStyle: { backgroundColor: headerColor },
+    headerTintColor: '#fff',
+    headerTitleStyle: {
+      fontWeight: 'bold',
+    },
+  };
+
+  stringifyCurrentProfile = (myProfile) => {
+    const myProfileCompressed = compressProfile(myProfile);
+    const jsonString = JSON.stringify(myProfileCompressed);
+    this.props.storeFriendsProfileJSONString(jsonString);
+  };
+
+  componentDidMount () {
+    // const { navigation } = this.props;
+    // this.stringifyCurrentProfile(navigation.getParam('contact'));
+
+    console.log('IN FRIEND PROFILE ', this.props.friendsProfileInView)
+    const { navigation } = this.props;
+    this.stringifyCurrentProfile(navigation.getParam('contact'));
+  };
+  
+  render() {
+    const { navigation } = this.props;
+    let contact = navigation.getParam('contact');
+
+
+    const { fName, lName, occupation, currentLocation: loc, email, birthplace  } = contact.personal;
+    const { facebook, instagram, twitter, blog } = contact.social;
+    const { linkedIn, github, cv, website } = contact.networking;
+    const {photo} = contact;
+    return (
+      
+      <View style={styles.container}>
+        <View style={styles.profileWrapper}>
+          <Avatar
+            avatarStyle={{ borderWidth: 2, borderColor: accentColorShadow }}
+            xlarge
+            rounded
+            source={{ uri: photo }}
+            onPress={() => console.log("Works!")}
+          />
+          <Text style={styles.name}> {`${fName} ${lName}`} </Text>
+          <Text style={styles.jobtitle}> {`${occupation}`} </Text>
+          <Text style={styles.details}>
+            Lives in - {`${loc.country}, ${loc.place}`}
+          </Text>
+          {/* <Text style={styles.miniInfo}>from <Text style={styles.category}>{` ${birthplace.country}`} </Text> </Text> */}
+
+          <TouchableOpacity style={styles.yellowButtonRound}>
+            <Icon git status
+              type='entypo'
+              name='link'
+              size={36}
+              color="black"
+              onPress={() => this.props.navigation.navigate('ShareFriendsQRCodeView',  { contact: contact })}
+            />
+          </TouchableOpacity>
+          
+        </View>
+
+        <View style={styles.iconsContainer}>
+
+          <Text style={styles.category}> Networking </Text>
+
+          <ScrollView horizontal contentContainerstyle={styles.iconsScrollView}>
+
+            <Icon iconStyle={styles.icons}
```

Comment:
There are 3 components that share this collection of icons. It's a good idea to modularize that.